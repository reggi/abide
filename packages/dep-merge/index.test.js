import {depMerge, unDepMerge} from './index'
import mockFs from 'mock-fs'
import fsExtra from 'fs-extra'
import fsMz from 'mz/fs'
const fs = {...fsExtra, ...fsMz}

const examplePackage = {
  'devDependencies': {
    'a': '*',
    'b': '*',
    'c': '*'
  },
  'dependencies': {
    'doe': '*',
    'rae': '*',
    'mi': '*'
  }
}

beforeEach(() => {
  mockFs({
    '/no-module': {},
    '/invalid-package': {
      'package.json': {}
    },
    '/my-module': {
      'package.json': JSON.stringify(examplePackage)
    }
  })
})

afterEach(async () => {
  mockFs.restore()
})

test('dep-merge pass empty dir', async () => {
  const results = await depMerge('/no-module')
  expect(results).toEqual({
    dirPath: '/no-module',
    endsInPackageJson: null,
    pkgPath: '/no-module/package.json',
    fileExists: false
  })
})

test('dep-merge pass dir with dir package.json', async () => {
  const results = await depMerge('/invalid-package')
  expect(results).toEqual({
    dirPath: '/invalid-package',
    endsInPackageJson: null,
    pkgPath: '/invalid-package/package.json',
    fileExists: false
  })
})

test('dep-merge', async () => {
  const results = await depMerge('/my-module')
  const dir = await fs.readdir('/my-module')
  expect(results).toEqual({
    'dirPath': '/my-module',
    'endsInPackageJson': null,
    'pkgPath': '/my-module/package.json',
    'fileExists': true,
    'packageContent': {
      'devDependencies': {
        'a': '*',
        'b': '*',
        'c': '*'
      },
      'dependencies': {
        'doe': '*',
        'rae': '*',
        'mi': '*'
      }
    },
    'newPackageJson': {
      'temporary-file:deletion-prevention': 'temporary-file:deletion-prevention',
      'devDependencies': {
        'a': '*',
        'b': '*',
        'c': '*'
      },
      'dependencies': {
        'doe': '*',
        'rae': '*',
        'mi': '*',
        'a': '*',
        'b': '*',
        'c': '*'
      }
    },
    'mvPath': '/my-module/original_package.json'
  })
  expect(dir).toEqual([ 'original_package.json', 'package.json' ])
})

test('dep-merge passing in dir', async () => {
  const results = await depMerge('/my-module')
  const dir = await fs.readdir('/my-module')
  expect(results).toMatchSnapshot()
  expect(dir).toEqual([ 'original_package.json', 'package.json' ])
  const umergeResult = await unDepMerge('/my-module')
  const dirAfterUnMerge = await fs.readdir('/my-module')
  const pkg = await fs.readJson('/my-module/package.json')
  expect(umergeResult).toMatchSnapshot()
  expect(dirAfterUnMerge).toEqual([ 'package.json' ])
  expect(pkg).toEqual(examplePackage)
})

test('dep-merge passing in pkg', async () => {
  const results = await depMerge('/my-module/package.json')
  const dir = await fs.readdir('/my-module')
  expect(results).toMatchSnapshot()
  expect(dir).toEqual([ 'original_package.json', 'package.json' ])
  const umergeResult = await unDepMerge('/my-module/package.json')
  const dirAfterUnMerge = await fs.readdir('/my-module')
  const pkg = await fs.readJson('/my-module/package.json')
  expect(umergeResult).toMatchSnapshot()
  expect(dirAfterUnMerge).toEqual([ 'package.json' ])
  expect(pkg).toEqual(examplePackage)
})
