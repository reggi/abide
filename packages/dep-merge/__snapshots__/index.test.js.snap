// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dep-merge passing in dir 1`] = `
Object {
  "dirPath": "/my-module",
  "endsInPackageJson": null,
  "fileExists": true,
  "moveOg": undefined,
  "mvPath": "/my-module/original_package.json",
  "newPackageJson": Object {
    "dependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
      "doe": "*",
      "mi": "*",
      "rae": "*",
    },
    "devDependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
    },
    "temporary-file:deletion-prevention": "temporary-file:deletion-prevention",
  },
  "outputJson": undefined,
  "packageContent": Object {
    "dependencies": Object {
      "doe": "*",
      "mi": "*",
      "rae": "*",
    },
    "devDependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
    },
  },
  "pkgPath": "/my-module/package.json",
}
`;

exports[`dep-merge passing in dir 2`] = `
Object {
  "dirPath": "/my-module",
  "endsInPackageJson": null,
  "moveOg": undefined,
  "mvPath": "/my-module/original_package.json",
  "newExists": true,
  "originalExists": true,
  "packageJsonContent": Object {
    "dependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
      "doe": "*",
      "mi": "*",
      "rae": "*",
    },
    "devDependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
    },
    "temporary-file:deletion-prevention": "temporary-file:deletion-prevention",
  },
  "pkgPath": "/my-module/package.json",
  "remove": undefined,
  "worthy": true,
}
`;

exports[`dep-merge passing in pkg 1`] = `
Object {
  "dirPath": "/my-module/package.json",
  "endsInPackageJson": Array [
    "package.json",
  ],
  "fileExists": true,
  "moveOg": undefined,
  "mvPath": "/my-module/original_package.json",
  "newPackageJson": Object {
    "dependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
      "doe": "*",
      "mi": "*",
      "rae": "*",
    },
    "devDependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
    },
    "temporary-file:deletion-prevention": "temporary-file:deletion-prevention",
  },
  "outputJson": undefined,
  "packageContent": Object {
    "dependencies": Object {
      "doe": "*",
      "mi": "*",
      "rae": "*",
    },
    "devDependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
    },
  },
  "pkgPath": "/my-module/package.json",
}
`;

exports[`dep-merge passing in pkg 2`] = `
Object {
  "dirPath": "/my-module/package.json",
  "endsInPackageJson": Array [
    "package.json",
  ],
  "moveOg": undefined,
  "mvPath": "/my-module/original_package.json",
  "newExists": true,
  "originalExists": true,
  "packageJsonContent": Object {
    "dependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
      "doe": "*",
      "mi": "*",
      "rae": "*",
    },
    "devDependencies": Object {
      "a": "*",
      "b": "*",
      "c": "*",
    },
    "temporary-file:deletion-prevention": "temporary-file:deletion-prevention",
  },
  "pkgPath": "/my-module/package.json",
  "remove": undefined,
  "worthy": true,
}
`;
